### ISSUE TITLE
Editing layer metadata in admin interface throws an error

### ISSUE THREAD

**Issue by virdiignis** [GeoNode/geonode#7735]
## Expected Behavior
Saving edited layer in django admin interface should succeed. 

## Actual Behavior
![image](https://user-images.githubusercontent.com/10606746/124453227-73ea7200-dd87-11eb-9080-6599f2fd5eee.png)

## Steps to Reproduce the Problem
  1. Prepare docker installation of GeoNode
  2. Upload layer
  3. Go to admin interface -> layers and edit uploaded layer
  4. Click save

## Specifications
  - GeoNode version: 3.2.0
  - Installation method: GeoNode Docker
  - Platform: Ubuntu Linux with docker
  - Additional details:
     Here is the content of the debug page: 
[TypeError.at._pl_admin_layers_layer_1_change_.zip](https://github.com/GeoNode/geonode/files/6763359/TypeError.at._pl_admin_layers_layer_1_change_.zip)

**Comment by virdiignis** [comment]
I found out that this issue is related to overridden **add** method in **_HierarchicalTagManager** class. It stopped to be compatible with django-taggit after they adjusted to changes in Django 2.2, so technically it's incompatible with Django 2.2 itself.

Anyways, it doesn't take keyword arguments that are passed by django-taggit code, from which the class inherits all other methods.

This is possibly related to #7750.

I have made a fix taking advantage of django-taggit's current shape of original **add** method. I'll check with tests if it's feasible for PR.

**Comment by afabiani** [comment]
I cannot reproduce this on 3.2.x branch so I guess this is a bug of the 3.2.0 release.

This might be already fixed on the upcoming 3.2.1

**Comment by afabiani** [comment]
nevermind, I was able to reproduce it

```
Internal Server Error: /en/admin/layers/layer/29/change/
Traceback (most recent call last):
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/options.py", line 606, in wrapper
    return self.admin_site.admin_view(view)(*args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/views/decorators/cache.py", line 44, in _wrapped_view_func
    response = view_func(request, *args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/sites.py", line 223, in inner
    return view(request, *args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1648, in change_view
    return self.changeform_view(request, object_id, form_url, extra_context)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/utils/decorators.py", line 45, in _wrapper
    return bound_method(*args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/utils/decorators.py", line 142, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1529, in changeform_view
    return self._changeform_view(request, object_id, form_url, extra_context)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1573, in _changeform_view
    self.save_related(request, form, formsets, not add)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/django/contrib/admin/options.py", line 1114, in save_related
    form.save_m2m()
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/dal/forms.py", line 141, in _save_m2m
    f.save_form_data(self.instance, cleaned_data[f.name])
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/taggit/managers.py", line 547, in save_form_data
    getattr(instance, self.name).set(*value)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/taggit/utils.py", line 124, in inner
    return func(self, *args, **kwargs)
  File "/home/geonode-vm-321/.virtualenvs/geonode/lib/python3.8/site-packages/taggit/managers.py", line 295, in set
    self.add(*new_objs, through_defaults=through_defaults, **kwargs)
TypeError: add() got an unexpected keyword argument 'through_defaults'
```
